osu.Game.Modes.Osu/Objects/OsuHitObjectParser.cs

```diff
@@ -4,39 +4,86 @@
 using System;
 using System.Collections.Generic;
 using System.Globalization;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using OpenTK;
+using osu.Game.Beatmaps;
 using osu.Game.Beatmaps.Samples;
 using osu.Game.Modes.Objects;
-using OpenTK;
 
 namespace osu.Game.Modes.Osu.Objects
 {
     public class OsuHitObjectParser : HitObjectParser
     {
-        public override HitObject Parse(string text)
+        public override HitObject Parse(Beatmap beatmap, string text)
         {
             string[] split = text.Split(',');
-            var type = (OsuHitObject.HitObjectType)int.Parse(split[3]);
+            if (split.Length < 5)
+                throw new ArgumentException("OsuHitObjects must have at least 5 values");
+
+            var type = (OsuHitObject.HitObjectType)Convert.ToInt32(split[3], NumberFormatInfo.InvariantInfo);

```

Comment:
are you sure? some of these values have decimals, so I think invariant culture is necessary (or did you just mean for integers?)